RPIPs/RPIP-43.md

```diff
@@ -60,14 +60,43 @@ Node operators can manage the set of validators in their megapool:
 
 ### Funds Management
 
-This subsection needs some work. I think it should say who can trigger
-distribution of funds out of the megapool, whether those funds are pending
-deposits, rewards, or remaining principal. This is a non-trivial question since
-funds do not necessarily belong to the node operator. There could be a link out
-to another RPIP with the details.
-
-- A Node Operator SHALL be able to distribute rewards from all validators in their megapool at once
-  - This MAY be temporarily blocked while validators are exiting or pending removal
+- This RPIP assumes the continued use of separate 32 Ether validators. Support for
+  EIP-7251 'Max EB' validators will require additional specification and development.
+- The megapool SHALL track the total capital ownership of the validator(s) between
+  the node operator and the rETH stakers.
+- The megapool MUST distinguish between capital funds present in the contract and
+  rewards funds at all times.
+- A Node Operator SHALL be able to distribute rewards from all validators in their
+  megapool at once.
+- Rewards distribution MAY be temporarily blocked while validators are exiting or
+  pending removal.
+- When the distribution function in the megapool is called:
+  - The node operator's share (withdrawn capital and node operator earnings plus
+    `node_operator_commission_share`) shall be sent to the node operator's
+    withdrawal address or held in the megapool as distributed but unclaimed node
+    operator funds.
+  - The rETH holder's share (withdrawn capital and reth earnings minus
+    `reth_commission`) SHALL be sent to the rETH contract or deposit pool depending
+    on the 'excess collateral' level in the rETH contract.
+  - The UARS rewards share (`voter_share` plus `surplus_share`) SHALL be sent to a
+    contract for further distribution via a merkle rewards distributor.
+- Anyone MAY perform a rewards distribution. Rewards distributions not performed
+  by the node operator SHALL leave the node operator's share in the megapools as
+  distributed but unclaimed node operator funds.
+- Anyone MAY perform a capital distribution. Distributions not performed by the
+  node operator SHALL be subject to a pDAO configured cooldown timer and SHALL
+  leave the node operator's share in the megapool as distributed, but unclaimed
+  node operator funds.
```

Comment:
```suggestion
- There SHALL be a reward distribution function in the megapool
  - When called, rewards from the node operator's bonded ETH SHALL be held in
    the megapool as unclaimed node operator funds.
  - When called, `node_operator_commission_share` of rewards SHALL be held in 
      the megapool as unclaimed node operator funds.
  - When called, `reth_share` of rewards SHALL be sent to the rETH contract
  - When called, `voter_share` of rewards SHALL be sent to a merkle rewards distributor contract
  - When called, `surplus_share` of rewards SHALL be sent to an appropriate distributor contract
  - This function SHALL allow any user to call it
  - If called by the node operator, this function SHOULD claim all unclaimed node operator funds
- There SHALL be a capital distribution function in the megapool
  - When called, capital from the node operator's bonded ETH that has been released from
    exited validators SHALL be held in the megapool as unclaimed node operator funds.
  - When called, capital borrowed from the protocol that has been released from
    exited validators SHALL be sent to the rETH contract.
  - This function SHALL allow any user to call it [DEFINE COOLDOWN]
  - If called by the node operator
    - This function SHOULD claim all unclaimed node operator funds
    - [BYPASS THE COOLDOWN THING DEFINED]
- There SHALL be a function to claim unclaimed node operator funds; when called
  - If unclaimed node operator funds >= penalties
    - unclaimed node operator funds SHALL be decreased by penalties
    - penalties SHALL be set to 0
  - If unclaimed node operator funds < penalties
    - unclaimed node operator funds SHALL be set to 0
    - penalties SHALL be decreased by unclaimed node operator funds
  - unclaimed node operator funds SHALL be transferred to the node operator's withdrawal address
  - unclaimed node operator funds SHALL be set to 0
```

The first section seems to posit a single distribution function; the last 2 bullets seem to posit separate functions for reward and capital distribution. I believe the reason for two is to enable the arb when there's a discount? That would require some amount of time where only the NO can do it. I think I will write it as two entirely.

Change discussion
- Separated the functions for rewards and capital
- I tried to remove overdefining (eg, sending to rETH contract or deposit pool, when rETH contract already has methods for overflowing; similar story on "unclaimed node operator funds" and withdrawal addy)
- MAY means that the bullet is optional to implement; I think it was used erroneously a place or two
- Tried to used "shares of rewards" more clearly.
- The UARs wording was made a bit broader -- burn and LP options are not using merkle rewards distributors.